src/platform/Darwin/DnssdContexts.cpp

```diff
@@ -516,32 +518,35 @@ void ResolveContext::DispatchSuccess()
     // ChipDnssdResolveNoLongerNeeded don't find us and try to also remove us.
     bool needDelete = MdnsContexts::GetInstance().RemoveWithoutDeleting(this);
 
-    for (auto & interface : interfaces)
-    {
-        auto & ips = interface.second.addresses;
+#if TARGET_OS_TV
+    // On tvOS, priotize results from en0, en1, ir0 in that order, if those
+    // interfaces are present, since those will generally have more up-to-date
+    // information.
+    static const unsigned int priorityInterfaceIndices[] = {
+        if_nametoindex("en0"),
+        if_nametoindex("en1"),
+        if_nametoindex("ir0"),
+    };
 
-        // Some interface may not have any ips, just ignore them.
-        if (ips.size() == 0)
+    for (auto interfaceIndex : priorityInterfaceIndices)
+    {
+        if (TryReportingResultsForInterfaceIndex(static_cast<uint32_t>(interfaceIndex)))
         {
-            continue;
+            if (needDelete)
+            {
+                MdnsContexts::GetInstance().Delete(this);
+            }
+            return;
         }
+    }
+#endif // TARGET_OS_TV
 
-        ChipLogProgress(Discovery, "Mdns: Resolve success on interface %" PRIu32, interface.first);
-
-        auto & service = interface.second.service;
-        auto addresses = Span<Inet::IPAddress>(ips.data(), ips.size());
-        if (nullptr == callback)
-        {
-            auto delegate = static_cast<CommissioningResolveDelegate *>(context);
-            DiscoveredNodeData nodeData;
-            service.ToDiscoveredNodeData(addresses, nodeData);
-            delegate->OnNodeDiscovered(nodeData);
-        }
-        else
+    for (auto & interface : interfaces)
```

Comment:
My only concern about this approach is that it will return the result for the same interface twice if it happens to be in the priority interface list.

Reply:
No, because we never reach this code if we found the thing in the priority interface list and it actually had IP addresses.  Note the "return" in the case when TryReportingResultsForInterfaceIndex returned true.

Reply:
Right. Thanks. :)